From b57d789f08ec9406c6e524238500ca4c783da50a Mon Sep 17 00:00:00 2001
From: Meron Cohen <mecohen@habana.ai>
Date: Tue, 03 Jun 2025 15:38:38 +0300
Subject: [PATCH] [SW-227617] common: fix readlink null terminate

The function readlink doesn't null terminate the target path buffer.
The function returns the number of bytes placed into this buffer or -1
in case of an error. Therefore, if there is no error, null terminate
the target path buffer. The side effect of it was returning wrong process
names (e.g., when running Deep Seek).

Change-Id: I4362d13605b3b5e43f4d632034b3d4f8b27c3c3d
Signed-off-by: Meron Cohen <mecohen@habana.ai>
---

diff --git a/app/hl-smi/info.cpp b/app/hl-smi/info.cpp
index 2db4901..2434394 100644
--- a/app/hl-smi/info.cpp
+++ b/app/hl-smi/info.cpp
@@ -1445,6 +1445,7 @@
 				if (r < 0)
 					continue;
 
+				points_to[r] = '\0';
 				for (i = 0; i< aips_cnt; i++) {
 					aip = &aips->aips_info.aip_info[i];
 					snprintf(buf, NAME_MAX, "%s/%s", device_path, aip->name);
